packages/block/src/block.ts

```diff
@@ -92,6 +94,24 @@ export class Block {
     return trie.root()
   }
 
+  /**
+   * Returns the requests trie root for an array of CLRequests
+   * @param requests - an array of CLRequests
+   * @param emptyTrie optional empty trie used to generate the root
+   * @returns a 32 byte Uint8Array representing the requests trie root
+   */
+  public static async genRequestsTrieRoot(requests: CLRequest[], emptyTrie?: Trie) {
+    const sortedRequests = requests?.sort((a, b) => {
+      if (a.type !== b.type) return a.type - b.type
+      return a.greaterThan(b) === true ? 1 : -1
+    })
+    const trie = emptyTrie ?? new Trie()
+    for (const [i, req] of sortedRequests.entries()) {
+      await trie.put(RLP.encode(i), req.serialize())
```

Comment:
It seems that this only sorts the types, but not the order of the requests of the same operation. Is this correct?

Reply:
It is to be defined by the operation, but I think the ordering of the same-type requests depend on whatever sorting algorithm is being used internally - right?

Reply:
We expect each request type to define a `greaterThan` function that we then use in a simple array `sort` method to take care of sorting.

Reply:
Oh right :man_facepalming: I failed to see that this was already in the sort method.